WPFHospitalEditor/Equipment.xaml.cs

```diff
@@ -0,0 +1,151 @@
+            gridRow1.Height = new GridLength(height);
+            DynamicGrid.RowDefinitions.Add(gridRow1);
+        }
+
+        private void createRowContent(int rowNumbers)
+        {
+                for (int i = 0; i < rowNumbers; i++)
+                {
+                    if (allEquipment.Count() == 0) break;
+                    setRowContent(i);
+                }
+            
+            insertData();
+            addCancelButton(rowNumbers);
+        }
+
+        private void setRowContent(int row)
+        {
+            labelContent.Add(allEquipment.ElementAt(row).Value.Name);
+            value.Add(allEquipment.ElementAt(row).Value.Quantity.ToString());
+        }
+
+        private void insertData()
+        {
+            for (int i = 0; i < labelContent.Count; i++)
+            {
+                insertLabel(i);
+                insertTextBox(i);
+            }
+        }
+
+        private void insertTextBox(int i)
+        {
+            TextBox textBox = new TextBox();
+            textBox.Text = value[i];
+            textBox.HorizontalContentAlignment = HorizontalAlignment.Center;
+            textBox.VerticalAlignment = VerticalAlignment.Center;
+            textBox.IsReadOnly = true;
+            Grid.SetRow(textBox, i + 2);
+            Grid.SetColumn(textBox, 2);
+            textBoxes.Add(textBox);
+            DynamicGrid.Children.Add(textBox);
+        }
+
+        private void insertLabel(int i)
+        {
+            Label label = new Label();
+            label.Content = labelContent[i];
+            label.HorizontalContentAlignment = HorizontalAlignment.Center;
+            label.VerticalAlignment = VerticalAlignment.Center;
+            label.FontSize = 20;
+            Grid.SetRow(label, i + 2);
+            Grid.SetColumn(label, 1);
+            labels.Add(label);
+            DynamicGrid.Children.Add(label);
+        }
+
+        private void addCancelButton(int rowNumbers)
+        {
+            Button cancel = new Button();
+            cancel.HorizontalAlignment = HorizontalAlignment.Left;
+            cancel.Click += Close_Click;
+            Grid.SetRow(cancel, rowNumbers + 2);
+            Grid.SetColumn(cancel, 1);
+            setButtonsCommonAttributes(cancel, "Cancel");
+        }
+
+        private void Close_Click(object sender, RoutedEventArgs e)
+        {
+            this.Close();
+        }
+
+        private void setButtonsCommonAttributes(Button button, string name)
+        {
+            button.BorderThickness = new Thickness(0);
+            button.Content = name;
+            button.VerticalAlignment = VerticalAlignment.Center;
+            button.Background = Brushes.SkyBlue;
+            button.Width = AllConstants.additionalInformationsbuttonWidth;
+            button.Height = AllConstants.additionalInformationsbuttonHeight;
+            button.Foreground = Brushes.White;
+            DynamicGrid.Children.Add(button);
+
+        }
+
+    }
+}
```

Comment:
All of these methods are identical to the AdditionalInformation.xaml.cs methods. It would be best to create a special component that will do all the above work, and in its constructor it is enough to specify only the data (probably string[] ) that should be used.